server/src/main/java/io/crate/expression/scalar/arithmetic/BinaryScalar.java

```diff
@@ -41,6 +41,7 @@ public BinaryScalar(BinaryOperator<T> func,
                         BoundSignature boundSignature,
                         DataType<T> type) {
         super(signature, boundSignature);
+        assert signature.hasFeature(Feature.NULLABLE) : "A BinaryScalar is NULLABLE by definition";
```

Comment:
I'm skeptical about this and the similar assertions, but on the other hand let's keep them and if there is a NON-NULLABLE function added we can remove them in the future.